## Risk Parameter Updates Aave V3 Polygon

_Updated as of block [17195267](https://etherscan.io/block/17195267) at 5/5/2023, 11:16:23 AM ET_

- ID: 216
- Proposer: [0xbC540e0729B732fb14afA240aA5A047aE9ba7dF0](https://etherscan.io/address/0xbC540e0729B732fb14afA240aA5A047aE9ba7dF0)
- Start Block: 17193612 (5/5/2023, 5:43:35 AM ET)
- End Block: 17212812 (5/8/2023, 2:37:48 AM ET)
- Targets: [0x158a6bC04F0828318821baE797f50B0A1299d45b](https://etherscan.io/address/0x158a6bC04F0828318821baE797f50B0A1299d45b#code)
- Executor: [0xEE56e2B3D491590B5b31738cC34d5232F378a8D5](https://etherscan.io/address/0xEE56e2B3D491590B5b31738cC34d5232F378a8D5) (Short executor)
- Simulation: [https://dashboard.tenderly.co/me/simulator/0d1eb949-7bbf-4f6b-b370-76d940c8bf97](https://dashboard.tenderly.co/me/simulator/0d1eb949-7bbf-4f6b-b370-76d940c8bf97)

### Subreports

- [PolygonBridgeExecution](/reports/Aave/0xEC568fffba86c094cf06b22134B23074DFE2252c/216_fx_1683299789_0.md)

<details>
  <summary>Proposal text</summary>

## Simple Summary

A proposal to adjust nine (9) total risk parameters, including Loan-to-Value, Liquidation Threshold, and Liquidation Bonus, across four (4) Aave V3 Polygon assets.

## Abstract

This proposal acts as a mandate from the Aave community to engage with Chaos for the work defined on the Aave governance forum [here](https://governance.aave.com/t/updated-proposal-chaos-labs-risk-simulation-platform/10025) and pre-approved by the Aave community via Snapshot [here](https://snapshot.org/#/aave.eth/proposal/0xad105e87d4df487bbe1daec2cd94ca49d1ea595901f5773c1804107539288b59).

## Motivation

Chaos Labs’ Parameter Recommendation Platform runs hundreds of thousands of agent-based off-chain and on-chain simulations to examine how different Aave V3 risk parameters configurations would behave under adverse market conditions - and find the optimal values to maximize protocol borrow usage while minimizing losses from liquidations and bad debt.

_Note: As a general guideline, we limit the proposed changes by +-3% for all parameters as a high/low bound for a given proposal. This ensures more controlled changes and allows us to analyze their effect on user behavior before recommending further amendments to the parameters if the optimal configuration is outside this range._

Please find more information on the parameter recommendation methodology [here](https://community.chaoslabs.xyz/aave/recommendations/methodology).

You can also view the simulation results and breakdown for the different assets by clicking on them on this [page](https://community.chaoslabs.xyz/aave/recommendations).

The output of our simulations reveals an opportunity to increase LTVs and LTs for WBTC, DAI, LINK and wMATIC on V3 Polygon, resulting in improved capital efficiency of the system, with no effect on the projected VaR (95th percentile of the protocol losses that will be accrued due to bad debt from under-collateralized accounts over 24 hours) and EVaR (Extreme VaR, the 99th percentile of the protocol losses that will be accrued due to bad debt from under-collateralized accounts over 24 hours)

Simulating all changes jointly yields a projected borrow increase of ~400,000$, with no increase in VaR and Extreme VaR compared to simulations with the current parameters.

## Specification

| Asset  | Parameter             | Current | Recommended | Change |
| ------ | --------------------- | ------- | ----------- | ------ |
| WBTC   | Liquidation Threshold | 75%     | 78%         | +3%    |
| WBTC   | Loan-to-Value         | 70%     | 73%         | +3%    |
| DAI    | Liquidation Threshold | 80%     | 81%         | +1%    |
| DAI    | Loan-to-Value         | 75%     | 76%         | +1%    |
| LINK   | Liquidation Threshold | 65%     | 68%         | +3%    |
| LINK   | Loan-to-Value         | 50%     | 53%         | +3%    |
| wMATIC | Liquidation Threshold | 70%     | 73%         | +3%    |
| wMATIC | Loan-to-Value         | 65%     | 68%         | +3%    |
| wMATIC | Liquidation Penalty   | 10%     | 7%          | -3%    |

# References

[Forum Post](https://governance.aave.com/t/arfc-chaos-labs-risk-parameter-updates-aave-v3-polygon-2023-04-23/12828)

[Test](https://github.com/bgd-labs/aave-proposals/blob/main/src/AaveV3PolRiskParams_20230423/AaveV3PolRiskParams_20230423_Test.t.sol)

[Proposal payload implementation](https://github.com/bgd-labs/aave-proposals/blob/main/src/AaveV3PolRiskParams_20230423/AaveV3PolRiskParams_20230423.sol)

## Copyright

Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

</details>

### Checks

#### Reports all state changes from the proposal ✅ Passed

Info:

- State changes:

```diff
# StateSender at `0x28e4F3a7f651294B9564800b2D01f35189A5bFbE`
@@ counter @@
- 2672443
+ 2672444
```

```diff
# Executor at `0xEE56e2B3D491590B5b31738cC34d5232F378a8D5`
@@ Slot `0x64e6c0494e94e8ceb4b4438ef268738a8d4e102ebe20fa029a9d3cff1fe553bd` @@
- "0x0000000000000000000000000000000000000000000000000000000000000001"
+ "0x0000000000000000000000000000000000000000000000000000000000000000"
```

#### Check stack trace of the proposal ✅ Passed

Info:

- There is no SELFDESTRUCT inside of delegated call

#### Reports all events emitted from the proposal ✅ Passed

Info:

- Events Emitted:
  - StateSender at `0x28e4F3a7f651294B9564800b2D01f35189A5bFbE`
    - `StateSynced(id: 2672444, contractAddress: 0x8397259c983751daf40400790063935a11afa28a, data: 0x000000000000000000000000ee56e2b3d491590b5b31738cc34d5232f378a8d5000000000000000000000000dc9a35b16db4e126cfedc41322b3a36454b1f7720000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000024000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000c53586aa2626094bd33c123794e34417ea877a3600000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000009657865637574652829000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001)`

#### Check all targets are verified on Etherscan ✅ Passed

Info:

- Targets:
  - 0x158a6bC04F0828318821baE797f50B0A1299d45b: Contract (verified) (CrosschainForwarderPolygon)

#### Check all touched contracts are verified on Etherscan ✅ Passed

Info:

- Touched address:
  - 0xD73a92Be73EfbFcF3854433A5FcbAbF9c1316073: EOA (verification not applicable)
  - 0xEC568fffba86c094cf06b22134B23074DFE2252c: Contract (verified) (AaveGovernanceV2)
  - 0xEE56e2B3D491590B5b31738cC34d5232F378a8D5: Contract (verified) (Executor)
  - 0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e: Contract (verified) (GovernanceStrategy)
  - 0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9: Contract (verified) (InitializableAdminUpgradeabilityProxy)
  - 0x96F68837877fd0414B55050c9e794AECdBcfCA59: Contract (verified) (AaveTokenV2)
  - 0x158a6bC04F0828318821baE797f50B0A1299d45b: Contract (verified) (CrosschainForwarderPolygon)
  - 0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2: Contract (verified) (FxRoot)
  - 0x28e4F3a7f651294B9564800b2D01f35189A5bFbE: Contract (verified) (StateSender)

#### Runs solc against the verified contracts ✅ Passed

Info:

-

<details>
<summary>View Details</summary>
<details>
<summary>View warnings for CrosschainForwarderPolygon at `0x158a6bC04F0828318821baE797f50B0A1299d45b`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
```

</details>

<details>
<summary>View warnings for StateSender at `0x28e4F3a7f651294B9564800b2D01f35189A5bFbE`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.5.11' not installed (set by SOLC_VERSION). Run `solc-select install 0.5.11`.

```

</details>

<details>
<summary>View warnings for InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.

```

</details>

<details>
<summary>View warnings for AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

```

</details>

<details>
<summary>View warnings for GovernanceStrategy at `0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

```

</details>

<details>
<summary>View warnings for FxRoot at `0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.3' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.3`.

```

</details>

</details>

#### Runs slither against the verified contracts ✅ Passed

Info:

-

<details>
<summary>View Details</summary>

<details>
<summary>Slither report for CrosschainForwarderPolygon at `0x158a6bC04F0828318821baE797f50B0A1299d45b`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
0x158a6bC04F0828318821baE797f50B0A1299d45b analyzed (2 contracts with 78 detectors), 0 result(s) found
```

</details>

<details>
<summary>Slither report for StateSender at `0x28e4F3a7f651294B9564800b2D01f35189A5bFbE`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.5.11' not installed (set by SOLC_VERSION). Run `solc-select install 0.5.11`.

Error in 0x28e4F3a7f651294B9564800b2D01f35189A5bFbE
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.5.11' not installed (set by SOLC_VERSION). Run `solc-select install 0.5.11`.


```

</details>

<details>
<summary>Slither report for InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.

Error in 0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.


```

</details>

<details>
<summary>Slither report for AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

Error in 0x96F68837877fd0414B55050c9e794AECdBcfCA59
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.


```

</details>

<details>
<summary>Slither report for GovernanceStrategy at `0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

Error in 0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.


```

</details>

<details>
<summary>Slither report for FxRoot at `0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.3' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.3`.

Error in 0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.3' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.3`.


```

</details>

</details>
